src/mantarray_file_manager/file_writer.py

```diff
@@ -156,3 +165,150 @@ def migrate_to_latest_version(
+    """
+    # pylint: disable-msg=too-many-locals
+    validate_int(
+        value=from_start,
+        allow_null=True,
+        minimum=0,
+    )
+    validate_int(value=from_end, allow_null=True, minimum=0)
+
+    if not from_start and not from_end:
+        raise UnsupportedArgumentError()
+
+    old_file_version = _get_format_version_of_file(file_path)
+
+    if old_file_version != CURRENT_HDF5_FILE_FORMAT_VERSION:
+        raise UnsupportedMantarrayFileVersionForTrimmingError(old_file_version)
+
+    working_directory = getcwd()
+    old_file = WellFile(file_path)
+    old_file_basename = ntpath.basename(file_path)[:-3]
+
+    # old metadata
+    old_h5_file = old_file.get_h5_file()
+    old_metadata_keys = set(old_h5_file.attrs.keys())
+    old_from_end = 0
+    old_from_start = 0
+
+    if str(IS_FILE_ORIGINAL_UNTRIMMED_UUID) in old_metadata_keys:
+        old_from_start = old_h5_file.attrs[str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)]
+        old_from_end = old_h5_file.attrs[str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID)]
+
+        old_metadata_keys.remove(str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID))
+        old_metadata_keys.remove(str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID))
+        old_metadata_keys.remove(str(IS_FILE_ORIGINAL_UNTRIMMED_UUID))
+
+        old_file_basename = old_file_basename.split("__trimmed")[0]
+
+    # finding amount to trim
+    old_raw_reference_data = old_file.get_raw_reference_reading()
+    old_tissue_data = old_file.get_raw_tissue_reading()
+
+    tissue_data_start_val = old_tissue_data[0][0]
+    tissue_data_last_val = old_tissue_data[0][-1]
+    tissue_data_start_index = _find_start_index(from_start, old_tissue_data[0])
+    tissue_data_last_index = _find_last_index(from_end, old_tissue_data)
+
+    actual_start_trimmed = (
+        old_tissue_data[0][tissue_data_start_index] - tissue_data_start_val
+    )
+    actual_end_trimmed = (
+        tissue_data_last_val - old_tissue_data[0][tissue_data_last_index]
+    )
+
+    reference_data_start_index = _find_start_index(
+        actual_start_trimmed, old_raw_reference_data[0]
+    )
+    reference_data_last_index = _find_last_index(
+        actual_end_trimmed, old_raw_reference_data
+    )
+
+    if (
+        reference_data_start_index >= reference_data_last_index
+        or tissue_data_start_index >= tissue_data_last_index
+    ):
+        raise TooTrimmedError(from_start, from_end)
+
+    if actual_start_trimmed != from_start:
+
+        print(  # allow-print
+            f"{actual_start_trimmed} centimilliseconds was trimmed from the start instead of {from_start}"
```

Comment:
if we have an important message to print, there is a way to mock print (something like `builtins.print`, you can check the desktop_app repo) so that you can assert that the message is appearing when you expect